code/obj/machinery/computer/bank_records.dm

```diff
@@ -264,6 +266,47 @@
 					if (transto == "Payroll") wagesystem.station_budget += amount
 					if (transto == "Shipping") wagesystem.shipping_budget += amount
 					if (transto == "Research") wagesystem.research_budget += amount
+				else if(href_list["bonus"])
+					if(!(world.time >= src.bonus_rate_limit_time))
+						boutput(usr, SPAN_ALERT("NT Regulations forbid issuing multiple staff incentives within five minutes."))
+						return
+
+					var/bonus = input(usr, "How many credits should we issue to each staff member?", "Issue Bonus", 100) as null|num
+					if(isnull(bonus)) return
+
+					bonus = ceil(clamp(bonus, 1, 999999))
+					var/bonus_total = (length(data_core.bank.records) * bonus)
+					if ( bonus_total > wagesystem.station_budget)
+						//Let the user know the budget is too small before they set the reason if we can
+						boutput(usr, SPAN_ALERT("Total bonus cost would be [bonus_total][CREDIT_SIGN], payroll budget is only [wagesystem.station_budget][CREDIT_SIGN]!"))
+						return
+
+					var/message = input(usr, "What is the reason for this staff bonus?", "Bonus Reason") as text
+					if(isnull(message) || message == "")
+						boutput(usr, SPAN_ALERT("NT Regulations require that the reason for issuing a staff bonus be recorded."))
+						return
+
+					//Something ain't right  if we enter either of these but it could be a coincidence
+					//Maybe someone stole the budget under our feet, or payroll was issued
+					if(isnull(bonus)|| isnull(bonus_total))
+						//No you really shouldn't be here
+						logTheThing(LOG_ADMIN, usr, "may have attempted href hacking Bank Computer. Do not take this as a guarantee.")
+						return
+					if(bonus_total > wagesystem.station_budget)
+						boutput(usr, SPAN_ALERT("Total bonus cost would be [bonus_total][CREDIT_SIGN], payroll budget is only [wagesystem.station_budget][CREDIT_SIGN]!"))
+						return
+
+					logTheThing(LOG_STATION, usr, "issued a bonus of [bonus][CREDIT_SIGN] ([bonus_total][CREDIT_SIGN] total) to the staff.")
+					src.bonus_rate_limit_time = world.time + (5 MINUTES)
+					command_announcement("[message]<br>Bonus of [bonus][CREDIT_SIGN] issued to all staff.", "Payroll Annoucment by [scan.registered] ([scan.assignment])")
+					wagesystem.station_budget = wagesystem.station_budget - bonus_total
+					for(var/datum/db_record/R as anything in data_core.bank.records)
+						if(R["job"] == "Clown")
+							//Tax the clown
+							R["current_money"] = (R["current_money"] + ceil((bonus / 2)))
+							return
```

Comment:
This will return the whole function and so stop running through all the records the moment it finds the clown, use an else instead

Reply:
oh dang! thanks for the catch!